### PR TITLE
feat: add e2e test

### PR DESCRIPTION
**Pull request by soulbird**
add e2e test

### CODE REVIEW

File: .github/workflows/e2e-test-ci.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        docker-compose -f ci/docker-compose.yml up -d
+        sleep 5
+
+    - name: build apisix-seed
+      run: |
+        go build
+
+    - name: startup apisix-seed
+      run: |
+        nohup ./apisix-seed &
```
Comment (spacewander) [inline .github/workflows/e2e-test-ci.yml:34]: Would be better to run apisix-seed as a docker too

File: test/e2e/regcenter/nacos_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package regcenter_test
+
+import (
+	"e2e/tools"
+	"e2e/tools/common"
+	. "github.com/onsi/ginkgo/v2"
```
Comment (spacewander) [inline test/e2e/regcenter/nacos_test.go:6]: Let's use `goimport` to solve the imports

File: test/e2e/tools/routes.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != 201 && resp.StatusCode != 200 {
+		panic("create route failed")
+	}
+	fmt.Println("create route successful: ", "/apisix/admin/routes/"+r.ID)
```
Comment (spacewander) [inline test/e2e/tools/routes.go]:
```suggestion
	fmt.Println("create route successful: /apisix/admin/routes/"+r.ID)
```

**Review by spacewander — COMMENTED** [review]
Shall we also provide a conf for seed?

File: .github/workflows/e2e-test-ci.yml (hunk 1 above)
Comment (spacewander) [reply .github/workflows/e2e-test-ci.yml:34]: So we can restart it just like other component.

Comment (soulbird) [reply .github/workflows/e2e-test-ci.yml:34]: > Would be better to run apisix-seed as a docker too

I think we should use the current version of the code for e2e testing. How to use docker to ensure that the code is updated in time?

File: test/e2e/regcenter/nacos_test.go (hunk 2 above)
Comment (soulbird) [reply test/e2e/regcenter/nacos_test.go:6]: This is the official recommendation of ginkgo, which can make the unit test code look more elegant

File: .github/workflows/e2e-test-ci.yml (hunk 1 above)
Comment (spacewander) [reply .github/workflows/e2e-test-ci.yml:34]: We can build it, like
https://github.com/apache/apisix/blob/6cd7b8d024d2b3da9db0a61dbff3b462059fa233/.github/workflows/chaos.yml#L42
https://github.com/apache/apisix-docker/blob/407610575b68b8764a3af787c9455e8bd7c1771f/example/docker-compose-alpine.yml#L22

File: test/e2e/regcenter/nacos_test.go (hunk 2 above)
Comment (spacewander) [reply test/e2e/regcenter/nacos_test.go:6]:
```
diff --git test/e2e/regcenter/nacos_test.go test/e2e/regcenter/nacos_test.go
index 250270a..bdd48d0 100644
--- test/e2e/regcenter/nacos_test.go
+++ test/e2e/regcenter/nacos_test.go
@@ -3,9 +3,10 @@ package regcenter_test
 import (
        "e2e/tools"
        "e2e/tools/common"
+       "time"
+
        . "github.com/onsi/ginkgo/v2"
        . "github.com/onsi/gomega"
-       "time"
 )

 var _ = Describe("Nacos", Ordered, func() {
```

Here is the result after sorting by goimport.

**Review by spacewander — APPROVED** [review]

**Pull request merged**